src/ContractInteract/EIP20Gateway.js

```diff
@@ -68,6 +97,244 @@ class EIP20Gateway {
     this.getValueTokenContract = this.getValueTokenContract.bind(this);
     this.isBountyAmountApproved = this.isBountyAmountApproved.bind(this);
     this.isStakeAmountApproved = this.isStakeAmountApproved.bind(this);
+    this.getStakeVault = this.getStakeVault.bind(this);
+    this.activateGateway = this.activateGateway.bind(this);
+    this.activateGatewayRawTx = this.activateGatewayRawTx.bind(this);
+  }
+
+  /**
+   * Setup for EIP20Gateway. Deploys EIP20Gateway and EIP20CoGateway,
+   * and links them.
+   *
+   * @param {Object} originWeb3 Web3 object.
+   * @param {Object} auxiliaryWeb3 Web3 object.
+   * @param {EIP20GatewaySetupConfig} gatewayConfig EIP20Gateway setup configuration.
+   * @param {EIP20CoGatewaySetupConfig} coGatewayConfig EIP20CoGateway setup configuration.
+   * @param {Object} originTxOptions Transaction options on Origin.
+   * @param {Object} auxiliaryTxOptions Transaction options on Auxiliary.
+   *
+   * @returns {Promise<EIP20Gateway>} Promise containing the EIP20Gateway instance that
+   *                                  has been set up.
+   */
+  static setup(
+    originWeb3,
+    auxiliaryWeb3,
+    gatewayConfig,
+    coGatewayConfig,
+    originTxOptions = {},
+    auxiliaryTxOptions = {},
+  ) {
+    if (!gatewayConfig) {
+      throw new Error('Mandatory parameter "gatewayConfig" missing. ');
+    }
+    if (!coGatewayConfig) {
+      throw new Error('Mandatory parameter "coGatewayConfig" missing. ');
+    }
+
+    gatewayConfig = gatewayConfig || {};
+    gatewayConfig.messageBus = gatewayConfig.messageBus || this.messageBus;
+    gatewayConfig.gatewayLib = gatewayConfig.gatewayLib || this.gatewayLib;
```

Comment:
`this` in a `static` method? What does `this` refer to here? 🤔

Reply:
Can probably be removed.